ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Protocol/Abstract.hs

```diff
@@ -170,13 +176,59 @@ class ( Show (ChainDepState   p)
   -- | We require that protocols support a @k@ security parameter
   protocolSecurityParam :: ConsensusConfig p -> SecurityParam
 
--- | Compare a candidate chain to our own
+-- | The chain order of some type; in the Consensus layer, this will always be
+-- the 'SelectView' of some 'ConsensusProtocol'.
 --
--- If both chains are equally preferable, the Ouroboros class of consensus
--- protocols /always/ sticks with the current chain.
-preferCandidate :: ConsensusProtocol p
-                => proxy      p
-                -> SelectView p  -- ^ Tip of our chain
-                -> SelectView p  -- ^ Tip of the candidate
-                -> Bool
-preferCandidate _ ours cand = cand > ours
+-- See 'preferCandidate' for the primary documentation.
+--
+-- Additionally, we require a total order on this type, such that eg different
+-- candidate chains can be sorted for prioritization.
+class Ord a => ChainOrder a where
+  type ChainOrderConfig a :: Type
```

Comment:
Out of interest: would use the `ChainOrderConfig` anywhere? Currently, none of the protocols rely on a config, do they?